packages/ra-core/src/auth/useLogoutIfAccessDenied.ts

```diff
@@ -40,21 +42,32 @@ const useLogoutIfAccessDenied = (): LogoutIfAccessDenied => {
     const logout = useLogout();
     const notify = useNotify();
     const logoutIfAccessDenied = useCallback(
-        (error?: any) =>
-            authProvider
-                .checkError(error)
-                .then(() => false)
-                .catch(e => {
-                    const redirectTo =
-                        e && e.redirectTo
-                            ? e.redirectTo
-                            : error && error.redirectTo
-                            ? error.redirectto
-                            : undefined;
-                    logout({}, redirectTo);
-                    notify('ra.notification.logged_out', 'warning');
-                    return true;
-                }),
+        (error?: any) => {
+            // Sometimes, a component might trigger multiple simultaneous
+            // dataProvider calls which all fail and call this function.
+            // To avoid having multiple notifications, we first verify if
+            // a checkError promise is already ongoing
+            if (!authCheckPromise) {
+                authCheckPromise = authProvider
```

Comment:
why do you store the entire Promise instead of just a boolean?

Reply:
Because this way, anyone interested by its result will receive it ?

Reply:
Hm, you're exposing a private API publicly without a real use case... I'm not comfortable with it.

Reply:
We were returning it before already. Nothing has changed. We just don't allow multiple simultaneous calls, that's all

Reply:
Gotcha, thanks for the explanation.